caffe2/contrib/tensorrt/tensorrt_tranformer.cc

```diff
@@ -163,39 +153,113 @@ void FillModelInfo(::ONNX_NAMESPACE::ModelProto* model) {
 }
 } // namespace
 
-OperatorDef TensorRTTransformer::BuildTrtOp(
-    const std::string& onnx_model_str,
+void BuildInitializationList(
+    Workspace* ws,
+    ::ONNX_NAMESPACE::GraphProto* g,
+    std::unordered_set<std::string>* initialization_list) {
+  const std::vector<string>& ws_blobs = ws->Blobs();
+
+  // Create a CUDA context and reuse it for potential tensor copies across
+  // devices
+  CUDAContext context;
+
+  for (const auto& s : ws_blobs) {
+    auto it = initialization_list->find(s);
+    if (it != initialization_list->end()) {
+      auto* init_tensor = g->add_initializer();
+      BlobToTensorProto(s, ws, &context, init_tensor);
+      initialization_list->erase(it);
+    }
+  }
+  CAFFE_ENFORCE(
+      initialization_list->empty(), "Unfulfilled initialization list");
+  for (const auto& t : g->initializer()) {
+    VLOG(2) << "Initializer: " << t.name();
+  }
+}
+
+void TensorRTTransformer::AddTrtOptions(
+    OperatorDef* op,
     const std::unordered_map<std::string, std::vector<int>>&
         output_size_hints) {
+  auto* max_batch_size_arg = op->add_arg();
```

Comment:
Using proto_utils, you can do `AddArgument("max_batch_size", max_batch_size_, op)` instead of these 3 lines

Reply:
Yeah, I tried that but every time I used that it gave me undefined symbols during linkage. :(